### DISCUSSION TITLE
What is the correct way of purging CDN when using custom provider

### DISCUSSION THREAD

**Discussion by balazsorban44** [vercel/next.js#14144] [Help] [answered]
When using a custom provider (ther than Vercel), and deploying a new version of the Next.js app, what are the required steps to correctly update the CDN?

Should I just purge everything, since the buildId is different on each builds? 

Or using https://nextjs.org/docs/api-reference/next.config.js/configuring-the-build-id would be ideal here? Or it may cause problems when part of the CDN expires and a request goes to origin, but that has already  changes for example?

In case I have to purge everything, I am wondering if that would mean that by using `getStaticProps` , all the previously cached props will not survive either? Is it maybe intentional?

**Comment by Timer** [comment, marked as answer]
Everything should be purged on a new deploy!
